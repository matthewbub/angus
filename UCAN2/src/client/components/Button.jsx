import clsx from "clsx";

/**
 * Button component.
 *
 * @param {Object} props - The component props.
 * @param {string} props.className - The class name for the button.
 * @param {ReactNode} props.children - The content of the button.
 * @param {...*} props.rest - Additional props for the button.
 * @returns {JSX.Element} The rendered Button component.
 */
const Button = ({ children, className, ...rest }) => (
  <button
    {...rest}
    className={clsx(
      "inline-flex items-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-600",
      className,
    )}
  >
    {children}
  </button>
);

export default Button;