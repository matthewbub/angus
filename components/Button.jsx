import clsx from "clsx";

/**
 * Button component.
 *
 * @param {Object} props - The component props.
 * @param {string} props.className - The class name for the button.
 * @param {ReactNode} props.children - The content of the button.
 * @param {...*} props.rest - Additional props for the button.
 * @returns {JSX.Element} The rendered Button component.
 */
const Button = ({
  children,
  className,
  styleType = 'primary',
  disabled = false,
  ...rest
}) => {
  const styleTypes = {
    primary: "bg-indigo-600 text-white",
    secondary: "bg-gray-600 text-white",
    danger: "bg-red-600 text-white",
    success: "bg-green-600 text-white",
    warning: "bg-yellow-600 text-white",
    info: "bg-blue-600 text-white",
    light: "bg-white text-gray-700",
    dark: "bg-gray-900 text-white",
  };

  const stateStyles = {
    primary: "hover:bg-indigo-700 focus:ring-indigo-500",
    secondary: "hover:bg-gray-700 focus:ring-gray-500",
    danger: "hover:bg-red-700 focus:ring-red-500",
    success: "hover:bg-green-700 focus:ring-green-500",
    warning: "hover:bg-yellow-700 focus:ring-yellow-500",
    info: "hover:bg-blue-700 focus:ring-blue-500",
    light: "hover:bg-gray-50 focus:ring-gray-500",
    dark: "hover:bg-gray-800 focus:ring-gray-500",
  };

  const baseStyles = "inline-flex items-center text-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm"

  return (
    <button
      {...rest}
      className={clsx(
        disabled ? "opacity-50 cursor-not-allowed" : stateStyles[styleType],
        baseStyles,
        styleTypes[styleType],
        className
      )}
    >
      {children}
    </button>
  )
};

export default Button;