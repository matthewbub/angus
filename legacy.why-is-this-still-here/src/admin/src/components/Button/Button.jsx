/**
 * Button component.
 *
 * @param {Object} props - The component props.
 * @param {string} props.className - The class name for the button.
 * @param {ReactNode} props.children - The content of the button.
 * @param {...*} props.rest - Additional props for the button.
 * @returns {JSX.Element} The rendered Button component.
 */
const Button = ({ children, className, ...rest }) => (
  <button {...rest} className='UCAN2-button'>
    {children}
  </button>
);

export default Button;