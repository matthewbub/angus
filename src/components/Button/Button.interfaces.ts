export interface ButtonProps {
  children: React.ReactNode;
  className?: string;
  styleType?: 'primary' | 'secondary' | 'danger' | 'success' | 'warning' | 'info' | 'light' | 'dark';
  disabled?: boolean;
  root?: 'button' | 'a';
  [x: string]: any;
}