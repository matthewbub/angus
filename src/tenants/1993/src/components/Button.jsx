import Link from 'next/link'
import clsx from 'clsx'

export const buttonBaseClasses = 'inline-flex rounded px-4 py-1.5 text-sm font-semibold transition'
export function Button({ invert, href, className, children, ...props }) {
  className = clsx(
    className,
    buttonBaseClasses,
    invert
      ? 'text-neutral-950'
      : 'bg-primary-500 text-white hover:bg-primary-600'
  )

  let inner = <span className="relative top-px">{children}</span>

  if (href) {
    return (
      <Link href={href} className={className} {...props}>
        {inner}
      </Link>
    )
  }

  return (
    <button className={className} {...props}>
      {inner}
    </button>
  )
}
