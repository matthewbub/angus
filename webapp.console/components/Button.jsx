import clsx from "clsx";

/**
 * Button component.
 *
 * @param {Object} props - The component props.
 * @param {string} props.className - The class name for the button.
 * @param {ReactNode} props.children - The content of the button.
 * @param {...*} props.rest - Additional props for the button.
 * @returns {JSX.Element} The rendered Button component.
 */
const Button = ({ children, className, styleType = 'primary', ...rest }) => {
  const styleTypes = {
    primary: "bg-indigo-600 hover:bg-indigo-700 focus:ring-indigo-500 text-white",
    secondary: "bg-gray-600 hover:bg-gray-700 focus:ring-gray-500 text-white",
    danger: "bg-red-600 hover:bg-red-700 focus:ring-red-500 text-white",
    success: "bg-green-600 hover:bg-green-700 focus:ring-green-500 text-white",
    warning: "bg-yellow-600 hover:bg-yellow-700 focus:ring-yellow-500 text-white",
    info: "bg-blue-600 hover:bg-blue-700 focus:ring-blue-500 text-white",
    light: "bg-white hover:bg-gray-50 focus:ring-gray-500 text-gray-700",
    dark: "bg-gray-900 hover:bg-gray-800 focus:ring-gray-500 text-white",
  };

  return (
    <button
      {...rest}
      className={clsx(
        styleTypes[styleType],
        "inline-flex items-center text-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2",
        className,
      )}
    >
      {children}
    </button>
  )
};

export default Button;