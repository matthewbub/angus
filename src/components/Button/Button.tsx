import clsx from "clsx";
import { FC } from "react";

import { ButtonProps } from './Button.interfaces';

const Button: FC<ButtonProps> = ({
  children,
  className,
  styleType = 'primary',
  disabled = false,
  root = 'button',
  ...rest
}) => {
  const styleTypes = {
    primary: "bg-indigo-600 txt1",
    secondary: "bg-gray-600 txt1",
    danger: "bg-red-600 txt1",
    success: "bg-green-600 txt1",
    warning: "bg-yellow-600 txt1",
    info: "bg-blue-600 txt1",
    light: "bg-white text-gray-700",
    dark: "bg-gray-900 txt1",
  } as const;

  const stateStyles = {
    primary: "hover:bg-indigo-700 focus:ring-indigo-500",
    secondary: "hover:bg-gray-700 focus:ring-gray-500",
    danger: "hover:bg-red-700 focus:ring-red-500",
    success: "hover:bg-green-700 focus:ring-green-500",
    warning: "hover:bg-yellow-700 focus:ring-yellow-500",
    info: "hover:bg-blue-700 focus:ring-blue-500",
    light: "hover:bg-gray-50 focus:ring-gray-500",
    dark: "hover:bg-gray-800 focus:ring-gray-500",
  } as const;

  const baseStyles = "inline-flex items-center text-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm" as const;

  if (root === 'a') {
    return (
      <a
        {...rest}
        className={clsx(
          disabled ? "opacity-50 cursor-not-allowed" : stateStyles[styleType],
          baseStyles,
          styleTypes[styleType],
          className
        )}
      >
        {children}
      </a>
    )
  }

  return (
    <button
      {...rest}
      className={clsx(
        disabled ? "opacity-50 cursor-not-allowed" : stateStyles[styleType],
        baseStyles,
        styleTypes[styleType],
        className
      )}
    >
      {children}
    </button>
  )
};

export default Button;