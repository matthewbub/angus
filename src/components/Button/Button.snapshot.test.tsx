import { render } from '@testing-library/react'
import Button from './Button'

it('renders Button unchanged', () => {
  const { container } = render(<Button>{'Hello, World!'}</Button>)
  expect(container).toMatchSnapshot()
})