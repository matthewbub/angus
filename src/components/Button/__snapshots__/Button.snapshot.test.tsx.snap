// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button unchanged 1`] = `
<div>
  <button
    class="hover:bg-teal-700 focus:ring-teal-500 inline-flex items-center text-center px-4 py-2 border border-transparent text-sm font-medium rounded-md shadow-sm bg-teal-600 txt1"
  >
    Hello, World!
  </button>
</div>
`;
